exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -56,7 +58,7 @@ def initialize(endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPOR
           raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || %w[gzip none].include?(compression)
 
           @uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
-                   URI("#{endpoint}/v1/traces")
+                   URI(File.join(endpoint, TRACES_PATH))
```

Comment:
We can just use `URI.join` here:
```
irb(main):002:0> uri = URI.join('https://localhost:1234/', 'v1/traces')
=> #<URI::HTTPS https://localhost:1234/v1/traces>
irb(main):003:0> uri = URI.join('https://localhost:1234', 'v1/traces')
=> #<URI::HTTPS https://localhost:1234/v1/traces>
```
```suggestion
                   URI.join(endpoint, 'v1/traces')
```